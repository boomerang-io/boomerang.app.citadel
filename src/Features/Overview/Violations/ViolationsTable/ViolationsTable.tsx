import React, { Component } from "react";
import moment from "moment";
import {
  Button,
  CodeSnippet,
  ComposedModal,
  DataTable,
  ModalFooter,
  ModalForm,
  ModalBody,
  Tag,
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module '@boo... Remove this comment to see the full error message
} from "@boomerang-io/carbon-addons-boomerang-react";
import copy from "copy-to-clipboard";
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module '@boo... Remove this comment to see the full error message
import { isAccessibleEvent } from "@boomerang-io/utils";
// @ts-expect-error ts-migrate(2307) FIXME: Cannot find module './violationsTable.module.scss'... Remove this comment to see the full error message
import styles from "./violationsTable.module.scss";

type Props = {
    violations?: any[];
};

type State = any;

export class ViolationsTable extends Component<Props, State> {

  annotationsRef = React.createRef();

  state = { isModalOpen: false, selectedPolicyIndex: null };
  headers = [
    {
      header: "Policy",
      key: "policyName",
    },
    {
      header: "Violations",
      key: "nbrViolations",
    },
    {
      header: "Failed Definition Types",
      key: "policyDefinitionTypes",
    },
    {
      header: "Reference ID",
      key: "referenceId",
    },
    {
      header: "Activity Date",
      key: "policyActivityCreatedDate",
    },
  ];

  subHeaders = [
    {
      header: "Metrics",
      key: "metric",
    },
    {
      header: "Message",
      key: "message",
    },
  ];

  renderCell = (rowIndex: any, cellIndex: any, value: any) => {
    const column = this.headers[cellIndex];
    switch (column.header) {
      case "Activity Date":
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        return <time className={styles.tableTextarea}>{value ? moment(value).format("MMMM DD, YYYY") : "---"}</time>;
      case "Failed Definition Types":
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        return <p className={styles.tableTextarea}>{value && value.length ? value.join(", ") : "---"}</p>;
      default:
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        return <p className={styles.tableTextarea}>{value || "---"}</p>;
    }
  };

  renderSubRow = (row: any) => {
    const { violations } = this.props;
    // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
    const currentViolation = violations.find((violation) => violation.id === row.id);
    if (currentViolation && currentViolation.violations.length > 0)
      // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
      return currentViolation.violations.map((violation: any) => <div className={styles.subRow}>
        {this.subHeaders.map((cell) => (
          // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
          <div key={cell.key} className={`${styles.tableCell} ${styles[cell.key]}`}>
            {this.renderDetail(cell.key, violation[cell.key])}
          </div>
        ))}
      </div>);
    else {
      return (
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        <div className={styles.subRow}>
          {this.subHeaders.map((cell) => (
            // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
            <div key={cell.key} className={`${styles.tableCell} ${styles[cell.key]}`}>
              {this.renderDetail(cell.key, "---")}
            </div>
          ))}
        </div>
      );
    }
  };

  renderDetail = (key: any, value: any) => {
    switch (key) {
      case "metric":
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        return <p className={styles.detailMetrics}>{value}</p>;
      case "message":
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        return <p className={styles.detailMessage}>{value}</p>;
      default:
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        return <p className={styles.tableTextarea}>{value || "---"}</p>;
    }
  };

  handleRowClick = (row: any) => {
    this.setState({
      isModalOpen: true,
      selectedPolicyIndex: row,
    });
  };

  render() {
    const { violations } = this.props;
    const { TableContainer, Table, TableHead, TableRow, TableBody, TableCell, TableHeader } = DataTable;
    // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
    const selectedViolation = violations[this.state.selectedPolicyIndex];
    // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
    return <>
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <ComposedModal
        isOpen={this.state.isModalOpen}
        composedModalProps={{ shouldCloseOnOverlayClick: true }}
        onCloseModal={() => this.setState({ isModalOpen: false })}
        modalHeaderProps={{ title: selectedViolation?.policyName }}
      >
        {() => (
          // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
          <ModalForm>
            {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
            <ModalBody>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <h2 className={styles.modalSectionTitle}>{`Violations (${
                selectedViolation?.violations?.length ?? 0
              })`}</h2>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <ul className={styles.modalLabels}>
                {(selectedViolation?.violations ?? [])?.map(({
                  metric,
                  message
                }: any, index: any) => (
                  // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
                  <li key={index}>
                    {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
                    <span>{metric} : </span>
                    {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
                    <span style={{ fontStyle: "italic" }}>{message}</span>
                  </li>
                ))}
              </ul>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <h2 className={styles.modalSectionTitle}>Failed Definition Types</h2>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <p>
                {selectedViolation?.policyDefinitionTypes && selectedViolation?.policyDefinitionTypes.length
                  ? selectedViolation?.policyDefinitionTypes.join(", ")
                  : "---"}
              </p>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <h2 className={styles.modalSectionTitle}>Labels</h2>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <ul className={styles.modalLabels}>
                {Object.entries(selectedViolation?.labels ?? {})?.map((entry) => (
                  // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
                  <li key={entry[0]}>
                    {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
                    <Tag type="purple">
                      {entry[0]}
                      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
                      <span>:</span>
                      {entry[1]}
                    </Tag>
                  </li>
                ))}
              </ul>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <h2 className={styles.modalSectionTitle}>Reference Link</h2>
              {selectedViolation?.referenceLink ? (
                // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
                <a href={selectedViolation.referenceLink} alt="Reference link">
                  {selectedViolation.referenceLink}
                </a>
              ) : (
                // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
                <p>No reference link provided</p>
              )}

              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <h2 className={styles.modalSectionTitle}>Reference ID</h2>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <p>{selectedViolation?.referenceId ?? "---"}</p>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <h2 className={styles.modalSectionTitle}>Created Date</h2>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <time>
                {selectedViolation?.createdDate
                  ? moment(selectedViolation.createdDate).format("MMMM DD, YYYY")
                  : "---"}
              </time>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <h2 className={styles.modalSectionTitle}>Annotations</h2>
              {selectedViolation?.annotations ? (
                // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
                <CodeSnippet
                  copyButtonDescription="Copy annotations to clipboard"
                  onClick={() => copy(JSON.stringify([selectedViolation?.annotations]))}
                  type="multi"
                >
                  {JSON.stringify([selectedViolation?.annotations], null, 1)}
                </CodeSnippet>
              ) : (
                "---"
              )}
            </ModalBody>
            {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
            <ModalFooter>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <Button onClick={() => this.setState({ isModalOpen: false })} data-testid="close-violation-detail">
                Close
              </Button>
            </ModalFooter>
          </ModalForm>
        )}
      </ComposedModal>
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <DataTable
        rows={violations}
        headers={this.headers}
        isSortable={true}
        render={({
          rows,
          headers,
          getHeaderProps,
          getRowProps
        }: any) => (
          // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
          <TableContainer>
            {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
            <Table isSortable className={styles.tableContainer} useZebraStyles={false}>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <TableHead>
                {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
                <TableRow className={styles.tableHeadRow}>
                  {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
                  {headers.map((header: any) => <TableHeader
                    key={header.key}
                    {...getHeaderProps({ header, className: `${styles.tableHeader} ${styles[header.key]}` })}
                  >
                    {header.header}
                  </TableHeader>)}
                </TableRow>
              </TableHead>
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <TableBody className={styles.tableBody}>
                {rows.map((row: any, rowIndex: any) => (
                  // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
                  <TableRow
                    className={styles.tableBodyRow}
                    key={row.id}
                    {...getRowProps({ row })}
                    onClick={() => this.handleRowClick(rowIndex)}
                    onKeyDown={(e: any) => isAccessibleEvent(e) && this.handleRowClick(rowIndex)}
                    tabIndex={0}
                    data-testid="violations-table-row"
                  >
                    {row.cells.map((cell: any, cellIndex: any) => (
                      // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
                      <TableCell key={cellIndex} style={{ padding: "0" }}>
                        {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
                        <div className={styles.tableCell}>{this.renderCell(rowIndex, cellIndex, cell.value)}</div>
                      </TableCell>
                    ))}
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </TableContainer>
        )}
      />
    </>;
  }
}

export default ViolationsTable;
