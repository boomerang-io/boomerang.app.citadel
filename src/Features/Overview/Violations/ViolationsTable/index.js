import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import moment from "moment";
import {
  Button,
  CodeSnippet,
  ComposedModal,
  DataTable,
  ModalFooter,
  ModalForm,
  ModalBody,
  Tag,
} from "@boomerang-io/carbon-addons-boomerang-react";
import copy from "copy-to-clipboard";
import { isAccessibleEvent } from "@boomerang-io/utils";
import styles from "./violationsTable.module.scss";

export class ViolationsTable extends Component {
  static propTypes = {
    violations: PropTypes.array,
  };

  annotationsRef = React.createRef();

  state = { isModalOpen: false, selectedPolicyIndex: null };
  headers = [
    {
      header: "Policy",
      key: "policyName",
    },
    {
      header: "Violations",
      key: "nbrViolations",
    },
    {
      header: "Failed Definition Types",
      key: "policyDefinitionTypes",
    },
    {
      header: "Reference ID",
      key: "referenceId",
    },
    {
      header: "Activity Date",
      key: "policyActivityCreatedDate",
    },
  ];

  subHeaders = [
    {
      header: "Metrics",
      key: "metric",
    },
    {
      header: "Message",
      key: "message",
    },
  ];

  renderCell = (rowIndex, cellIndex, value) => {
    const column = this.headers[cellIndex];
    switch (column.header) {
      case "Activity Date":
        return <time className={styles.tableTextarea}>{value ? moment(value).format("MMMM DD, YYYY") : "---"}</time>;
      case "Failed Definition Types":
        return <p className={styles.tableTextarea}>{value && value.length ? value.join(", ") : "---"}</p>;
      default:
        return <p className={styles.tableTextarea}>{value || "---"}</p>;
    }
  };

  renderSubRow = (row) => {
    const { violations } = this.props;
    const currentViolation = violations.find((violation) => violation.id === row.id);
    if (currentViolation && currentViolation.violations.length > 0)
      return currentViolation.violations.map((violation) => (
        <div className={styles.subRow}>
          {this.subHeaders.map((cell) => (
            <div key={cell.key} className={`${styles.tableCell} ${styles[cell.key]}`}>
              {this.renderDetail(cell.key, violation[cell.key])}
            </div>
          ))}
        </div>
      ));
    else {
      return (
        <div className={styles.subRow}>
          {this.subHeaders.map((cell) => (
            <div key={cell.key} className={`${styles.tableCell} ${styles[cell.key]}`}>
              {this.renderDetail(cell.key, "---")}
            </div>
          ))}
        </div>
      );
    }
  };

  renderDetail = (key, value) => {
    switch (key) {
      case "metric":
        return <p className={styles.detailMetrics}>{value}</p>;
      case "message":
        return <p className={styles.detailMessage}>{value}</p>;
      default:
        return <p className={styles.tableTextarea}>{value || "---"}</p>;
    }
  };

  handleRowClick = (row) => {
    this.setState({
      isModalOpen: true,
      selectedPolicyIndex: row,
    });
  };

  render() {
    const { violations } = this.props;
    const { TableContainer, Table, TableHead, TableRow, TableBody, TableCell, TableHeader } = DataTable;
    const selectedViolation = violations[this.state.selectedPolicyIndex];
    return (
      <>
        <ComposedModal
          isOpen={this.state.isModalOpen}
          composedModalProps={{ shouldCloseOnOverlayClick: true }}
          onCloseModal={() => this.setState({ isModalOpen: false })}
          modalHeaderProps={{ title: selectedViolation?.policyName }}
        >
          {() => (
            <ModalForm>
              <ModalBody>
                <h2 className={styles.modalSectionTitle}>{`Violations (${
                  selectedViolation?.violations?.length ?? 0
                })`}</h2>
                <ul className={styles.modalLabels}>
                  {(selectedViolation?.violations ?? [])?.map(({ metric, message }, index) => (
                    <li key={index}>
                      <span>{metric} : </span>
                      <span style={{ fontStyle: "italic" }}>{message}</span>
                    </li>
                  ))}
                </ul>
                <h2 className={styles.modalSectionTitle}>Failed Definition Types</h2>
                <p>
                  {selectedViolation?.policyDefinitionTypes && selectedViolation?.policyDefinitionTypes.length
                    ? selectedViolation?.policyDefinitionTypes.join(", ")
                    : "---"}
                </p>
                <h2 className={styles.modalSectionTitle}>Labels</h2>
                <ul className={styles.modalLabels}>
                  {Object.entries(selectedViolation?.labels ?? {})?.map((entry) => (
                    <li key={entry[0]}>
                      <Tag type="purple">
                        {entry[0]}
                        <span>:</span>
                        {entry[1]}
                      </Tag>
                    </li>
                  ))}
                </ul>
                <h2 className={styles.modalSectionTitle}>Reference Link</h2>
                {selectedViolation?.referenceLink ? (
                  <a href={selectedViolation.referenceLink} alt="Reference link">
                    {selectedViolation.referenceLink}
                  </a>
                ) : (
                  <p>No reference link provided</p>
                )}

                <h2 className={styles.modalSectionTitle}>Reference ID</h2>
                <p>{selectedViolation?.referenceId ?? "---"}</p>
                <h2 className={styles.modalSectionTitle}>Created Date</h2>
                <time>
                  {selectedViolation?.createdDate
                    ? moment(selectedViolation.createdDate).format("MMMM DD, YYYY")
                    : "---"}
                </time>
                <h2 className={styles.modalSectionTitle}>Annotations</h2>
                {selectedViolation?.annotations ? (
                  <CodeSnippet
                    copyButtonDescription="Copy annotations to clipboard"
                    onClick={() => copy(JSON.stringify([selectedViolation?.annotations]))}
                    type="multi"
                  >
                    {JSON.stringify([selectedViolation?.annotations], null, 1)}
                  </CodeSnippet>
                ) : (
                  "---"
                )}
              </ModalBody>
              <ModalFooter>
                <Button onClick={() => this.setState({ isModalOpen: false })} data-testid="close-violation-detail">
                  Close
                </Button>
              </ModalFooter>
            </ModalForm>
          )}
        </ComposedModal>
        <DataTable
          rows={violations}
          headers={this.headers}
          isSortable={true}
          render={({ rows, headers, getHeaderProps, getRowProps }) => (
            <TableContainer>
              <Table className={styles.tableContainer} sortable={"true"} useZebraStyles={false}>
                <TableHead>
                  <TableRow className={styles.tableHeadRow}>
                    {headers.map((header) => (
                      <TableHeader
                        {...getHeaderProps({ header, className: `${styles.tableHeader} ${styles[header.key]}` })}
                      >
                        {header.header}
                      </TableHeader>
                    ))}
                  </TableRow>
                </TableHead>
                <TableBody className={styles.tableBody}>
                  {rows.map((row, rowIndex) => (
                    <Fragment key={row.id}>
                      <TableRow
                        className={styles.tableBodyRow}
                        key={row.id}
                        {...getRowProps({ row })}
                        onClick={() => this.handleRowClick(rowIndex)}
                        onKeyDown={(e) => isAccessibleEvent(e) && this.handleRowClick(rowIndex)}
                        tabIndex={0}
                        data-testid="violations-table-row"
                      >
                        {row.cells.map((cell, cellIndex) => (
                          <TableCell key={cell.id} style={{ padding: "0" }}>
                            <div className={styles.tableCell}>{this.renderCell(rowIndex, cellIndex, cell.value)}</div>
                          </TableCell>
                        ))}
                      </TableRow>
                    </Fragment>
                  ))}
                </TableBody>
              </Table>
            </TableContainer>
          )}
        />
      </>
    );
  }
}

export default ViolationsTable;
