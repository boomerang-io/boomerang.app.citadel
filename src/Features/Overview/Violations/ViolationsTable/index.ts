// @ts-expect-error ts-migrate(6142) FIXME: Module './ViolationsTable' was resolved to 'C:/Use... Remove this comment to see the full error message
export { default } from "./ViolationsTable"