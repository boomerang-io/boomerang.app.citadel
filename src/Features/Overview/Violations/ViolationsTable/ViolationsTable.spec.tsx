import React from "react";
import ViolationsTable from "./index";
const violations = [
    {
        id: "1",
        policyId: "5cd49adff6ea74a9bb6adef4",
        policyName: "Tyson's Policy",
        ciStageId: "5cb123bdf6ea74a9bb5425b6",
        ciStageName: "dev",
        ciComponentId: "5cdb4c15f6ea74a9bb9cfdba",
        ciComponentName: "boomerang.service.ci",
        ciComponentVersionId: "5cdb4ca3f6ea74a9bb9cf251",
        ciComponentVersionName: "6.4.9-175",
        policyActivityCreatedDate: 1558453998666,
        violations: 3
    }
];
describe("ViolationsTable --- Snapshot", () => {
    it("Capturing Snapshot of ViolationsTable", async () => {
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        const { baseElement } = (global as any).rtlRender(<ViolationsTable violations={violations}/>);
        expect(baseElement).toMatchSnapshot();
    });
});
