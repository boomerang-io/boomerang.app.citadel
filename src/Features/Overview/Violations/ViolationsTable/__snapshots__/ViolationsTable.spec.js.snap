// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViolationsTable --- Snapshot Capturing Snapshot of ViolationsTable 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="bx--data-table-container"
    >
      <table
        class="bx--data-table tableContainer bx--data-table--no-border"
        sortable="true"
      >
        <thead>
          <tr
            class="tableHeadRow"
          >
            <th
              aria-sort="none"
              class="tableHeader policyName"
              scope="col"
            >
              <button
                class="tableHeader policyName bx--table-sort"
              >
                <span
                  class="bx--table-header-label"
                >
                  Policy
                </span>
                <svg
                  aria-label="Sort rows by this header in ascending order"
                  class="bx--table-sort__icon"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 4L6 14 7.41 15.41 15 7.83 15 30 17 30 17 7.83 24.59 15.41 26 14 16 4z"
                  />
                </svg>
                <svg
                  aria-label="Sort rows by this header in ascending order"
                  class="bx--table-sort__icon-unsorted"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                  />
                </svg>
              </button>
            </th>
            <th
              aria-sort="none"
              class="tableHeader nbrViolations"
              scope="col"
            >
              <button
                class="tableHeader nbrViolations bx--table-sort"
              >
                <span
                  class="bx--table-header-label"
                >
                  Violations
                </span>
                <svg
                  aria-label="Sort rows by this header in ascending order"
                  class="bx--table-sort__icon"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 4L6 14 7.41 15.41 15 7.83 15 30 17 30 17 7.83 24.59 15.41 26 14 16 4z"
                  />
                </svg>
                <svg
                  aria-label="Sort rows by this header in ascending order"
                  class="bx--table-sort__icon-unsorted"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                  />
                </svg>
              </button>
            </th>
            <th
              aria-sort="none"
              class="tableHeader policyDefinitionTypes"
              scope="col"
            >
              <button
                class="tableHeader policyDefinitionTypes bx--table-sort"
              >
                <span
                  class="bx--table-header-label"
                >
                  Failed Definition Types
                </span>
                <svg
                  aria-label="Sort rows by this header in ascending order"
                  class="bx--table-sort__icon"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 4L6 14 7.41 15.41 15 7.83 15 30 17 30 17 7.83 24.59 15.41 26 14 16 4z"
                  />
                </svg>
                <svg
                  aria-label="Sort rows by this header in ascending order"
                  class="bx--table-sort__icon-unsorted"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                  />
                </svg>
              </button>
            </th>
            <th
              aria-sort="none"
              class="tableHeader referenceId"
              scope="col"
            >
              <button
                class="tableHeader referenceId bx--table-sort"
              >
                <span
                  class="bx--table-header-label"
                >
                  Reference ID
                </span>
                <svg
                  aria-label="Sort rows by this header in ascending order"
                  class="bx--table-sort__icon"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 4L6 14 7.41 15.41 15 7.83 15 30 17 30 17 7.83 24.59 15.41 26 14 16 4z"
                  />
                </svg>
                <svg
                  aria-label="Sort rows by this header in ascending order"
                  class="bx--table-sort__icon-unsorted"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                  />
                </svg>
              </button>
            </th>
            <th
              aria-sort="none"
              class="tableHeader policyActivityCreatedDate"
              scope="col"
            >
              <button
                class="tableHeader policyActivityCreatedDate bx--table-sort"
              >
                <span
                  class="bx--table-header-label"
                >
                  Activity Date
                </span>
                <svg
                  aria-label="Sort rows by this header in ascending order"
                  class="bx--table-sort__icon"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 4L6 14 7.41 15.41 15 7.83 15 30 17 30 17 7.83 24.59 15.41 26 14 16 4z"
                  />
                </svg>
                <svg
                  aria-label="Sort rows by this header in ascending order"
                  class="bx--table-sort__icon-unsorted"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M27.6 20.6L24 24.2 24 4 22 4 22 24.2 18.4 20.6 17 22 23 28 29 22zM9 4L3 10 4.4 11.4 8 7.8 8 28 10 28 10 7.8 13.6 11.4 15 10z"
                  />
                </svg>
              </button>
            </th>
          </tr>
        </thead>
        <tbody
          aria-live="polite"
          class="tableBody"
        >
          <tr
            class="tableBodyRow"
            data-testid="violations-table-row"
            tabindex="0"
          >
            <td
              style="padding: 0px;"
            >
              <div
                class="tableCell"
              >
                <p
                  class="tableTextarea"
                >
                  Tyson's Policy
                </p>
              </div>
            </td>
            <td
              style="padding: 0px;"
            >
              <div
                class="tableCell"
              >
                <p
                  class="tableTextarea"
                >
                  ---
                </p>
              </div>
            </td>
            <td
              style="padding: 0px;"
            >
              <div
                class="tableCell"
              >
                <p
                  class="tableTextarea"
                >
                  ---
                </p>
              </div>
            </td>
            <td
              style="padding: 0px;"
            >
              <div
                class="tableCell"
              >
                <p
                  class="tableTextarea"
                >
                  ---
                </p>
              </div>
            </td>
            <td
              style="padding: 0px;"
            >
              <div
                class="tableCell"
              >
                <time
                  class="tableTextarea"
                >
                  January 01, 2019
                </time>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    class="bx--bmrg-modal-portal"
  />
</body>
`;
